libtenzir/src/index.cpp

```diff
@@ -530,16 +530,15 @@ caf::error index_state::load_from_disk() {
         if (auto error = extract_partition_synopsis(part_path, synopsis_path))
           return error;
       }
-      auto chunk = chunk::mmap(synopsis_path);
-      if (!chunk)
-        return chunk.error();
-      const auto* ps_flatbuffer
-        = fbs::GetPartitionSynopsis(chunk->get()->data());
+      TRY(auto chunk, chunk::mmap(synopsis_path));
+      TRY(const auto ps_flatbuffer,
+          flatbuffer<fbs::PartitionSynopsis>::make(std::move(chunk)));
```

Comment:
Since this now runs the FlatBuffers verifier, we should probably check how much startup time is negatively impacted.

Reply:
I tried to do that, but waiting for the catalog to finish loading is not possible from a pipeline as is, and we'd have to add special code for that. Given that it should only cause a marginal increase in index startup latency I think we should move ahead without this test.